common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java

```diff
@@ -444,6 +447,57 @@ public static void validateUpdateUser(Request userRequest) {
     }
   }
 
+  public static void externalIdsValidation(Request userRequest) {
+    if (userRequest.getRequest().containsKey(JsonKey.EXTERNAL_IDS)
+        && (null != userRequest.getRequest().get(JsonKey.EXTERNAL_IDS))) {
+      if (!(userRequest.getRequest().get(JsonKey.EXTERNAL_IDS) instanceof List)) {
+        throw new ProjectCommonException(
+            ResponseCode.dataTypeError.getErrorCode(),
+            ProjectUtil.formatMessage(
+                ResponseCode.dataTypeError.getErrorMessage(), JsonKey.EXTERNAL_IDS, JsonKey.LIST),
+            ERROR_CODE);
+      } else {
+        List<Map<String, String>> externalIds = null;
+        List<String> operationList = null;
+        try {
+          externalIds =
+              (List<Map<String, String>>) userRequest.getRequest().get(JsonKey.EXTERNAL_IDS);
+          operationList =
+              Arrays.asList(ProjectUtil.OperationType.values())
+                  .stream()
+                  .map(s -> s.name().toLowerCase())
+                  .collect(Collectors.toList());
+        } catch (Exception ex) {
+          throw new ProjectCommonException(
+              ResponseCode.dataTypeError.getErrorCode(),
+              ProjectUtil.formatMessage(
+                  ResponseCode.dataTypeError.getErrorMessage(),
+                  JsonKey.EXTERNAL_IDS,
+                  "List<Map<String,String>>"),
+              ERROR_CODE);
+        }
+
+        List<String> operationTypeList = new ArrayList<String>(operationList);
+        externalIds
+            .stream()
+            .forEach(
+                s -> {
+                  if (StringUtils.isNotBlank(s.get(JsonKey.OPERATION))
+                      && (!operationTypeList.contains((s.get(JsonKey.OPERATION)).toLowerCase()))) {
+                    throw new ProjectCommonException(
```

Comment:
What happens if "operation" value is not specified? we want to use "Add" as the default value.